src/main/java/mil/dds/anet/utils/DaoUtils.java

```diff
@@ -334,4 +336,25 @@ public static Instant getCurrentMinute() {
     final ZonedDateTime bom = now.truncatedTo(ChronoUnit.MINUTES);
     return bom.toInstant();
   }
+
+  public static List<String> getEmailAddressesBasedOnPreferences(List<? extends Person> people,
+      String preference) {
+    // Load people preferences
+    CompletableFuture.allOf(people.stream()
+        .map(a -> a.loadPreferences(ApplicationContextProvider.getEngine().getContext()))
+        .toArray(CompletableFuture<?>[]::new)).join();
+
+    // Load email addresses only of people who did not opt out of report emails
+    CompletableFuture.allOf(people.stream()
+        .filter(person -> person.getPreferences().stream()
+            .noneMatch(pref -> pref.getPreference().getName().equals(preference)
```

Comment:
`noneMatch()`? Surely you just want to filter for the ones who have this value as TRUE?